packages/stark-ui/src/modules/minimap/components/minimap.component.ts

```diff
@@ -0,0 +1,173 @@
+	public isDisplayedMenu: boolean;
+
+	public constructor(@Inject(STARK_LOGGING_SERVICE) public logger: StarkLoggingService, private ngZoneService: NgZone) {}
+
+	public ngOnInit(): void {
+		this.visibleItems = _cloneDeep(this.parentVisibleItems) || [];
+		this.items = _cloneDeep(this.parentItems) || [];
+
+		this.isDisplayedMenu = false;
+		this.attachWindowClickHandler();
+	}
+
+	/**
+	 * Component lifecycle hook
+	 */
+	public ngOnChanges(changesObj: SimpleChanges): void {
+		if (changesObj["parentItems"] && !_isEqual(this.parentItems, changesObj["parentItems"].previousValue)) {
+			this.logger.debug(componentName + ": items changed... Recreating items config...", changesObj["parentItems"]);
+			this.items = _cloneDeep(this.parentItems);
+		}
+
+		if (changesObj["parentVisibleItems"] && !_isEqual(this.parentVisibleItems, changesObj["parentVisibleItems"].previousValue)) {
+			this.logger.debug(
+				componentName + ": visibleItems changed... Recreating visibleItems config...",
+				changesObj["parentVisibleItems"]
+			);
+			this.visibleItems = _cloneDeep(this.parentVisibleItems);
+		}
+	}
+
+	/**
+	 * Component lifecycle hook
+	 */
+	public ngOnDestroy(): void {
+		this.logger.debug(componentName + ": removing clickHandler");
+		this.removeWindowClickHandler();
+	}
+
+	/**
+	 * Retrieve the label of an item
+	 * @param StarkMinimapItemProperties - the item whose label we want to retrieve
+	 * @returns string - the label of the item
+	 */
+	public getItemLabel(item: StarkMinimapItemProperties): string {
+		if (item.label) {
+			return item.label;
+		}
+
+		return item.name;
+	}
+
+	/**
+	 * Attach a handler when a click is performed on the window
+	 */
+	public attachWindowClickHandler(): void {
+		this.windowClickHandler = (event: Event) => {
+			if (this.isDisplayedMenu) {
+				const parentElement: Element | undefined = StarkDOMUtil.searchParentElementByClass(
+					<Element>event.target,
+					"stark-minimap-dropdown-toggle"
+				);
+
+				if (!parentElement) {
+					this.ngZoneService.run(() => this.toggleDropdownMenu());
+				}
+			}
+		};
+
+		window.addEventListener("click", this.windowClickHandler);
+	}
+
+	/**
+	 * remove the handler attached to a window
+	 */
+	public removeWindowClickHandler(): void {
+		window.removeEventListener("click", this.windowClickHandler);
+	}
+
+	/**
+	 * Return true/false if the given item is already visible or if the priority (if specified) for such item is not "hidden"
+	 * Otherwise, the item is considered to be hidden by default
+	 */
+	public isItemVisible(item: StarkMinimapItemProperties): boolean {
+		return this.visibleItems instanceof Array ? this.visibleItems.indexOf(item.name) > -1 : false;
+	}
+
+	/**
+	 * Display / Hide the dropdown menu
+	 */
+	public toggleDropdownMenu(): void {
+		this.isDisplayedMenu = !this.isDisplayedMenu;
+	}
+
+	/**
+	 * triggers the show/hide event on an item
+	 * @param StarkMinimapItemProperties : item - the item to show/hide
```

Comment:
I think this JSDoc should be like this: `@param item - The item to show/hide`... because the type is already implicit in the code ;)